src/FSharp.Core/fslib-extra-pervasives.fsi

```diff
@@ -585,3 +594,9 @@ namespace Microsoft.FSharp.Core.CompilerServices
         ///
         /// <returns>The provided method definition corresponding to the given static parameter values</returns>
         abstract ApplyStaticArgumentsForMethod : methodWithoutArguments:MethodBase * methodNameWithArguments:string * staticArguments:obj[] -> MethodBase
+
+    type ITypeProvider3 =
+        abstract ApplyStaticArguments : context:ITypeProviderDiagnosticsContext * typeWithoutArguments:Type * typePathWithArguments:string[] * staticArguments:obj[] -> Type 
+
+        abstract ApplyStaticArgumentsForMethod:
+            context: ITypeProviderDiagnosticsContext * methodWithoutArguments: MethodBase * methodNameWithArguments: string * staticArguments: obj[] -> MethodBase
```

Comment:
@dsyme, I couldn't extend `ITypeProvider` as you had suggested, because TPs that rely on the old version of the interface in their ProvidedTypes.fs would fail to load. There's `ITypeProvider2` that allows to tack on additional functionality, so I opted for the same approach.

Reply:
Now that I test backwards compatibility with VS which includes FSharp.Core 7.0 without these additions, `ITypeProvider3` can't be loaded and the TP doesn't work. I'm at a loss here.

Reply:
Yes, but I don't see how that would solve the problem.